src/libexpr/primops.cc

```diff
@@ -51,11 +50,15 @@ StringMap EvalState::realiseContext(const PathSet & context)
         auto c = NixStringContextElem::parse(*store, c_);
         std::visit(overloaded {
             [&](const NixStringContextElem::Built & b) {
+                auto ctxS = store->printStorePath(b.drvPath);
                 drvs.push_back(DerivedPath::Built {
                     .drvPath = b.drvPath,
                     .outputs = OutputsSpec::Names { b.output },
                 });
                 ensureValid(b.drvPath);
+                if (evalSettings.logAllIFD) {
+                    printInfo("%1% importing from derivation %2% via %3%", positions[pos], ctxS, reason);
```

Comment:
To make this useful for automated processing (which I assume is the goal), this should be an activity/result with its own `act*` / `res*` code and the position/reason/store path as separate fields.

"Importing from derivation" is a misnomer here since "reason" might not be an import.